src/record/record.js

```diff
@@ -345,10 +324,13 @@ Record.prototype._recoverRecord = function( remoteVersion, remoteData, message )
  * @private
  * @returns {void}
  */
-Record.prototype._onRecordRecovered = function( remoteVersion, error, data ) {
+Record.prototype._onRecordRecovered = function( remoteVersion, remoteData, error, data ) {
 	if( !error ) {
 		this.version = remoteVersion;
-		this.set( data );
+		if ( data !== this._$data && !utils.deepEquals( data, this._$data ) ) {
```

Comment:
@yasserf: Understood. I think all current behaviour is maintained and the tests pass. I've already made a separate issue for this problem.